### ISSUE TITLE
Drop down menu does not scroll using a screen reader

### ISSUE THREAD

**Issue by kennethj** [flutter/flutter#84719]
## Steps to Reproduce

Using a screen reader run the code below:
1. Open the menu by tapping on the `DropdownButton`.
2. Swipe to navigate to through the list
3. When reaching the end of the visible list focus will leave the list instead of scrolling to the next item

```dart
import 'package:flutter/material.dart';

main() => runApp(MyApp());

class MyApp extends StatefulWidget {
  @override
  MyAppState createState() => MyAppState();
}

class MyAppState extends State<MyApp> with WidgetsBindingObserver {
  @override
  Widget build(context) => MaterialApp(title: 'Semantics bug demo app', home: HomePage());
}

class HomePage extends StatefulWidget {
  @override
  _HomePageState createState() => _HomePageState();
}

class _HomePageState extends State<HomePage> {
  int selected = 0;
  @override
  Widget build(context) => Scaffold(
        appBar: AppBar(title: Text("testing")),
        body: Center(
          child: DropdownButton(
            items: List.generate(100, (index) => DropdownMenuItem<int>(child: Text("Item #$index "), value: index)),
            value: selected,
            onChanged: (value) => setState(() => selected = value),
          ),
        ),
      );
}
```

**Expected results:** 
As a screen reader user, I expect the semantic focus to go to the next item of the list of items until it reaches the end.

**Actual results:** 
However, when the focus reaches the end of the visible items in the scrollable region it announces "Dismiss", ignoring most of the items in the list. There is no indication to the screenreader that there are more items or that the list is scrollable.

**Comment by darshankawar** [comment]
Thanks for the detailed report and code sample. I see the same behavior. When dropdown is tapped to show list of menu items and voiceover is ON, it reads all menu items one by one that are displayed / visible on screen. Upon reaching last visible item, it reads `dismiss`.

<img width="350" alt="Screenshot 2021-06-17 at 1 23 54 PM" src="https://user-images.githubusercontent.com/67046386/122355129-48861b80-cf6f-11eb-81e0-3433af709587.png">


<details>
<summary>flutter doctor -v</summary>

```console
/Users/dhs/Documents/Fluttersdk/flutter/bin/flutter doctor --verbose
[✓] Flutter (Channel stable, 2.2.2, on Mac OS X 10.15.4 19E2269 darwin-x64, locale en-GB)
    • Flutter version 2.2.2 at /Users/dhs/Documents/Fluttersdk/flutter
    • Framework revision d79295af24 (6 days ago), 2021-06-11 08:56:01 -0700
    • Engine revision 91c9fc8fe0
    • Dart version 2.13.3

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.3, Build version 12C33
    • CocoaPods version 1.10.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] VS Code (version 1.56.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.21.0

[✓] Connected device (3 available)
    • Darshan's iphone (mobile) • 21150b119064aecc249dfcfe05e259197461ce23 • ios            • iOS 14.4.1
    • macOS (desktop)           • macos                                    • darwin-x64     • Mac OS X 10.15.4 19E2269 darwin-x64
    • Chrome (web)              • chrome                                   • web-javascript • Google Chrome 91.0.4472.106

• No issues found!

```
</details>

**Comment by chunhtai** [comment]
There are two issues, and both of them are tricky

1. the shrinkwrap list view does not return the correct semantics clip which causes the hidden child to be dropped from the semantics tree
2. The animatedOpacity of the widget marks the individual item to need semantics update, but not the parent of the item, which causes the item to rebuild its semantics node without its parent because the semantics node ancestor chain is build top down. which makes it to be marked not hidden

**Comment by chunhtai** [comment]
reverted due to internal test failure
